secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/jws/signer/JwsSigner.java

```diff
@@ -0,0 +1,174 @@
+ *                     .as(secretsProvider(heap));
+ *             final String signingKeyId = config.get("signingKeyId").required().asString();
+ *             final String kid = config.get("kid").as(evaluatedWithHeapProperties()).required().asString();
+ *             final String algorithm = config.get("algorithm").required().asString();
+ *             return new CustomSapiJwsSigner(secretsProvider, signingKeyId, kid, algorithm);
+ *         }
+ *     }
+ * }
+ * }</pre>
+ * Heap configuration <br/>
+ * <p/>
+ * Mandatory fields:
+ * <ul>
+ *     <li>secretsProvider: The SecretsProvider object to query for the 'signingKeyId' in the keystore</li>
+ *     <li>signingKeyId: The signing key id name to identify the private key in the keystore to sign a JWT</li>
+ *     <li>kid: Key ID to build the JWT header, used to validate the signature via JWKs</li>
+ *     <li>algorithm: The name of the algorithm to use to sign the JWT</li>
+ * </ul>
+ * Example config:<br/>
+ * The heap configuration must be included as an object in the proper config.json descriptor.
+ * <pre>{@code
+ * {
+ *     "comment": "Default payload signer",
+ *     "name": "CustomSigner",
+ *     "type": "[[package].CustomSigner] or [alias name]",
+ *     "config": {
+ *         "algorithm": "PS256",
+ *         "signingKeyId": "jwt.signer",
+ *         "kid": "&{ig.ob.aspsp.signing.kid}",
+ *         "secretsProvider": "SecretsProvider-ASPSP"
+ *     }
+ * }
+ * }</pre>
+ * <p>
+ * <h4>Script filter configuration</h4>
+ * <pre>{@code
+ * {
+ *   "comment": "Sign messages",
+ *   "name": "CustomSigner",
+ *   "type": "ScriptableFilter",
+ *   "config": {
+ *     "type": "application/x-groovy",
+ *     "file": "CustomScript.groovy",
+ *     "args": {
+ *       "signer": "${heap['CustomSigner']}"
+ *     }
+ *   }
+ * }
+ * }</pre>
+ * </p>
+ * <p>
+ * <h4>Script logic</h4>
+ * <pre>{@code
+ * signer.sign(payloadMap, critClaims)
+ *                 .then(signedJwt -> {
+ *                     logger.debug("result {}", signedJwt)
+ *                     // process the signed JWT (string) result
+ *                     return result
+ *                 }, jwsSignerException -> { // JwsSignerException handler
+ *                     logger.error("Signature fails: {}", jwsSignerException.getMessage())
+ *                     response.status = Status.INTERNAL_SERVER_ERROR
+ *                     response.entity = "{ \"error\":\"" + jwsSignerException.getMessage() + "\"}"
+ *                     return newResultPromise(response)
+ *                 })
+ * }</pre>
+ * </p>
+ * <br/>
+ * <p>
+ * See as example:
+ * <ul>
+ *     <li>Signer implementation class: com.forgerock.sapi.gateway.jws.signer.CompactSerializationJwsSigner</li>
+ *     <li>Configuration descriptor: config.json</li>
+ *     <li>Route: 68-ob-aggregated-polling</li>
+ *     <li>Script: SignEventsResponse</li>
+ * </ul> as example<br/>
+ * </p>
```

Comment:
I think this is better suited to a wiki entry too. Otherwise we end up with too much noise in the javadoc. For the interface I just want to know what it is/how I use it and the method level contracts.

@BohoCode what are your thoughts?

Reply:
Deleted to move it to the wiki as generic extension topic.